.gitlab-ci.yml

```diff
@@ -1,20 +1,58 @@
 stages:
+  - publish_build_pending
   - build
+  - publish_build_status
 
-trigger_testbed:
-  stage: build
+variables:
+  GIT_SUBMODULE_STRATEGY: recursive
+  DRIVER_NAME: embedded-sht
+  
+.publish_build_status:
   image:
     name: registry.gitlab.sensirion.lokal/sensirion/docker/docker-ubuntu:18.04-1.2.0
-  tags:
-    - docker
-    - linux
+  tags: [docker, linux]
   script:
   - >
     curl
-    --request POST
-    --form ref=master
-    --form token=$CI_JOB_TOKEN
-    --form "variables[DRIVER_NAMES]=embedded-sht"
-    --form "variables[DRIVER_BRANCH]=$CI_COMMIT_REF_NAME"
-    --form "variables[DRIVER_COMMIT_SHA]=$CI_COMMIT_SHA"
-    https://gitlab/api/v4/projects/801/trigger/pipeline
+    -f
+    -H "Authorization: token $TESTBED_GITHUB_TOKEN"
+    -H "Content-Type: application/json"
+    -d "{\"state\":\"$PUBLISH_STATE\",\"description\":\"$PUBLISH_DESCRIPTION\",\"context\":\"ci/internal-ci\"}"
+    https://api.github.com/repos/sensirion/$DRIVER_NAME/statuses/$CI_COMMIT_SHA
+  except:
+    variables:
+      - $DRIVER_BRANCH =~ /^internal\//
+
+publish_pending:
+  extends: .publish_build_status
+  stage: publish_build_pending
+  variables:
+    PUBLISH_STATE: "pending"
+    PUBLISH_DESCRIPTION: "Waiting for internal tests to finish"
+
+test:
+  stage: build
+  image:
+    name: registry.gitlab.sensirion.lokal/sensirion/docker/docker-rpi-testbed:1.1.0
+  tags:
+    - mso-sw-testbed-2
+    - ch-sta-p-rp0002-DOCKER
+  script:
+    - git clone https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab/MSO-SW/drivers/testbed/i2c-mux-testbed.git
```

Comment:
For reproducibility we should specify a tag or revision here. Otherwise if there are changes to the i2c-mux-testbed project it may suddenly fail tests.